src/Common/src/Interop/Unix/System.Net.Http.Native/Interop.Initialization.cs

```diff
@@ -24,8 +24,18 @@ internal static class HttpInitializer
         static HttpInitializer()
         {
 #if !SYSNETHTTP_NO_OPENSSL
-            // CURL uses OpenSSL which me must initialize first to guarantee thread-safety
-            CryptoInitializer.Initialize();
+            string opensslVersion = Interop.Http.GetSslVersionDescription();
+            if (string.IsNullOrEmpty(opensslVersion) ||
+                opensslVersion.IndexOf("openssl/1.0", StringComparison.OrdinalIgnoreCase) != -1)
+            {
+                // CURL uses OpenSSL which me must initialize first to guarantee thread-safety
+                CryptoInitializer.Initialize();
+            }
+            else
+            {
+                // Some newer version of openssl used, which might result in segfaults.
+                throw new PlatformNotSupportedException(SR.Format(SR.net_http_unix_openssl_version, opensslVersion));
```

Comment:
This is a pretty serious regression.  We should just treat openssl/1.0 the way we'd currently treat nss... things work as long as you don't try doing custom cert handling.